codec/types/interface_registry.go

```diff
@@ -138,6 +148,22 @@ func (registry *interfaceRegistry) registerImpl(iface interface{}, typeURL strin
 		panic(fmt.Errorf("type %T doesn't actually implement interface %+v", impl, ityp))
 	}
 
+	// Check if we already registered something under the given typeURL. It's
+	// okay to register the same concrete type again, but if we are registering
+	// a new concrete type under the same typeURL, then we throw an error (here,
+	// we panic).
+	foundImplType, found := imap[typeURL]
+	if found && foundImplType != implType {
+		panic(
```

Comment:
sorry Robert, it's hard not to panic here. I think we need to change the signature of `RegisterInterfaces` everywhere on AppModuleBasic, if we don't want to panic.